internal/uvm/create.go

```diff
@@ -286,23 +315,46 @@ func Create(opts *UVMOptions) (*UtilityVM, error) {
 		}
 
 		if opts.ConsolePipe != "" {
-			hcsDocument.VirtualMachine.Chipset.UEFI.BootThis.OptionalData += " console=ttyS0,115200"
-			hcsDocument.VirtualMachine.Devices.COMPorts = &schema2.VirtualMachinesResourcesComPortsV2{Port1: opts.ConsolePipe}
+			vmDebugging = true
+			kernelArgs += " console=ttyS0,115200"
+			vm.Devices.COMPorts = &schema2.VirtualMachinesResourcesComPortsV2{Port1: opts.ConsolePipe}
 		}
 
 		if opts.EnableGraphicsConsole {
-			hcsDocument.VirtualMachine.Chipset.UEFI.BootThis.OptionalData += " console=tty"
-			hcsDocument.VirtualMachine.Devices.Keyboard = &schema2.VirtualMachinesResourcesKeyboardV2{}
-			hcsDocument.VirtualMachine.Devices.Rdp = &schema2.VirtualMachinesResourcesRdpV2{}
-			hcsDocument.VirtualMachine.Devices.VideoMonitor = &schema2.VirtualMachinesResourcesVideoMonitorV2{}
+			vmDebugging = true
+			kernelArgs += " console=tty"
+			vm.Devices.Keyboard = &schema2.VirtualMachinesResourcesKeyboardV2{}
+			vm.Devices.Rdp = &schema2.VirtualMachinesResourcesRdpV2{}
+			vm.Devices.VideoMonitor = &schema2.VirtualMachinesResourcesVideoMonitorV2{}
+		}
+
+		if !vmDebugging {
+			// Terminate the VM if there is a kernel panic.
+			kernelArgs += " panic=-1"
 		}
 
 		if opts.KernelBootOptions != "" {
-			hcsDocument.VirtualMachine.Chipset.UEFI.BootThis.OptionalData = hcsDocument.VirtualMachine.Chipset.UEFI.BootThis.OptionalData + fmt.Sprintf(" %s", opts.KernelBootOptions)
+			kernelArgs += " " + opts.KernelBootOptions
+		}
+
+		// Start GCS with stderr pointing to the vsock port created below in
+		// order to forward guest logs to logrus.
+		initArgs := fmt.Sprintf("/bin/vsockexec -e %d /bin/gcs -log-format json -loglevel %s",
```

Comment:
I didn't really consider usage like this. I figured we would always want all of them or none of them. I need to verify that the gcs only ever writes to stderr again

Reply:
We can still use vsockexec to open the bridge connection if you want, but yeah I think this should work fine for the log processing.

Reply:
Nah this should work. Just need to make sure GCS writes all log data to stdout

Reply:
You mean stderr.